packages/common/src/JsonRpcClient.ts

```diff
@@ -0,0 +1,195 @@
+import { Buffer } from 'buffer/';
+import {
+    AccountTransaction,
+    AccountTransactionSignature,
+    ConsensusStatus,
+    ContractAddress,
+    InstanceInfo,
+    NextAccountNonce,
+    TransactionStatus,
+    TransactionSummary,
+} from './types';
+import { AccountAddress } from './types/accountAddress';
+import Provider, { JsonRpcResponse } from './providers/provider';
+import { serializeAccountTransactionForSubmission } from './serialization';
+import { GtuAmount } from './types/gtuAmount';
+import { ModuleReference } from './types/moduleReference';
+import { buildJsonResponseReviver, intToStringTransformer } from './util';
+
+function handleResponse<R>(
+    res: JsonRpcResponse,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    transform: (res: any) => R = (result) => result
+): R | undefined {
+    if (res.error) {
+        throw new Error(res.error.code + ': ' + res.error.message);
+    } else if (res.result) {
+        return transform(res.result);
+    }
+    return undefined;
+}
+
+function transformJsonResponse(
+    jsonString: string,
+    reviver?: (this: unknown, key: string, value: unknown) => unknown,
+    transformer?: (json: string) => string
+) {
+    if (jsonString === 'null') {
+        return undefined;
+    }
+
+    if (transformer) {
+        const transformedJson = transformer(jsonString);
+        return JSON.parse(transformedJson, reviver);
+    }
+
+    return JSON.parse(jsonString, reviver);
+}
+
+export class JsonRpcClient {
+    provider: Provider;
+
+    constructor(provider: Provider) {
+        this.provider = provider;
+    }
+
+    async getNextAccountNonce(
+        accountAddress: AccountAddress
+    ): Promise<NextAccountNonce | undefined> {
+        const response = await this.provider.request('getNextAccountNonce', {
+            address: accountAddress.address,
+        });
+
+        const bigIntPropertyKeys: (keyof NextAccountNonce)[] = ['nonce'];
+
+        return handleResponse(response, (r) =>
+            transformJsonResponse(
+                JSON.stringify(r),
```

Comment:
This looks problematic. What happens if there is a `bigint` in `r` when doing this?

Reply:
Yeah, it was lossy, I've reworked the handling, to avoid a lossy parse before this.